osu.Game/Graphics/UserInterface/RollingCounter.cs

```diff
@@ -46,57 +47,71 @@ public abstract class RollingCounter<T> : Container, IHasAccentColour
         public virtual T DisplayedCount
         {
             get => displayedCount;
-
             set
             {
                 if (EqualityComparer<T>.Default.Equals(displayedCount, value))
                     return;
 
                 displayedCount = value;
-                DisplayedCountSpriteText.Text = FormatCount(value);
+                if (displayedCountSpriteText != null)
+                    displayedCountSpriteText.Text = FormatCount(value);
             }
         }
 
         public abstract void Increment(T amount);
 
+        private float textSize = 40f;
+
         public float TextSize
         {
-            get => DisplayedCountSpriteText.Font.Size;
-            set => DisplayedCountSpriteText.Font = DisplayedCountSpriteText.Font.With(size: value);
+            get => displayedCountSpriteText?.Font.Size ?? textSize;
+            set
+            {
+                if (TextSize == value)
+                    return;
+
+                textSize = value;
+                if (displayedCountSpriteText != null)
+                    displayedCountSpriteText.Font = displayedCountSpriteText.Font.With(size: value);
+            }
         }
 
+        private Color4 accentColour;
+
         public Color4 AccentColour
         {
-            get => DisplayedCountSpriteText.Colour;
-            set => DisplayedCountSpriteText.Colour = value;
+            get => displayedCountSpriteText?.Colour ?? accentColour;
+            set
+            {
+                if (AccentColour == value)
+                    return;
+
+                accentColour = value;
+                if (displayedCountSpriteText != null)
+                    displayedCountSpriteText.Colour = value;
+            }
         }
 
         /// <summary>
         /// Skeleton of a numeric counter which value rolls over time.
         /// </summary>
         protected RollingCounter()
         {
-            Children = new Drawable[]
-            {
-                DisplayedCountSpriteText = new OsuSpriteText { Font = OsuFont.Numeric }
-            };
-
-            TextSize = 40;
             AutoSizeAxes = Axes.Both;
 
-            DisplayedCount = Current.Value;
-
             Current.ValueChanged += val =>
             {
-                if (IsLoaded) TransformCount(displayedCount, val.NewValue);
+                if (IsLoaded)
+                    TransformCount(DisplayedCount, val.NewValue);
             };
         }
 
-        protected override void LoadComplete()
+        [BackgroundDependencyLoader]
+        private void load()
         {
-            base.LoadComplete();
-
-            DisplayedCountSpriteText.Text = FormatCount(Current.Value);
+            displayedCountSpriteText = CreateSpriteText();
+            displayedCountSpriteText.Text = FormatCount(displayedCount);
+            Child = displayedCountSpriteText;
```

Comment:
really confused as to why `displayedCount` and `DisplayedCount` are used interchangeably here - would argue that only the property should access its backing field directly and everything else should use the getter

Reply:
whoops I forgot checking this